syfertext/doc.py

```diff
@@ -154,8 +155,61 @@ def vector(self):
 
         return doc_vector
 
+    def get_vector(self, excluded_tokens: Dict = None):
+        """
+        Get document vector as an average of in-vocabulary token's vectors,
+        excluding token according to the excluded_tokens dictionary.
+
+        Args
+            excluded_tokens (Dict): A dictionary used to ignore tokens of the document based on values of their attributes,
+                                    the keys are the attributes names and they index lists values.
+                                        Example: {'attribute1_name' : [value1, value2],
+                                                    'attribute2_name': [v1, v2], ....}
+
+        Returns:
+          doc_vector: document vector ignoring excluded tokens
+        """
+
+        vectors = None
+
+        # Count the tokens that have vectors
+        vector_count = 0
+
+        for token in self:
+
+            # Get the vector of the token if one exists and if token is not excluded
+
+            include_token = True
+            if excluded_tokens is not None:
+                include_token = all(
+                    [
+                        getattr(token._, key) not in excluded_tokens[key]
+                        for key in excluded_tokens.keys()
+                        if hasattr(token._, key)
+                    ]
+                )
+
+            if token.has_vector and include_token:
+                # Increment the vector counter
+                vector_count += 1
+
+                # Cumulate token's vector by summing them
+                vectors = token.vector if vectors is None else vectors + token.vector
+
+        # If no tokens with vectors were found, just get the default vector(zeros)
+        if vector_count == 0:
+            doc_vector = self.vocab.vectors.default_vector
+        else:
+            # Create the final Doc vector
+            doc_vector = vectors / vector_count
+        return doc_vector
+
     def get_encrypted_vector(
-        self, *workers: BaseWorker, crypto_provider: BaseWorker = None, requires_grad: bool = True
+        self,
+        *workers: BaseWorker,
+        crypto_provider: BaseWorker = None,
+        requires_grad: bool = True,
+        excluded_tokens: Dict = None,
     ):
```

Comment:
Awesome, you have included the `excluded_tokens` in this method.

However, in the methods code you need also to enable using `self.get_vector` instead of `self.vector`.

Reply:
Yes, I'll do it in the next commit.